### PR TITLE
fix: editor and cocos-play issues

### PR DESCRIPTION
**Pull request by freeboy**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/root.ts [hunk 1]
```diff
@@ -342,7 +342,7 @@ export class Root {
      * @param width The new width of the window.
      * @param height The new height of the window.
      */
-    public resize (width: number, height: number) {
+    public resize (windowId: number, width: number, height: number) {
```
Comment (PPpro) [inline cocos/core/root.ts]: it's a breaking change for `root.resize` interface
please define the interface as `public resize (width: number, height: number, windowId?: number)`

**Review by PPpro — CHANGES_REQUESTED** [review]

File: cocos/core/root.ts (hunk 1 above)
Comment (freeboy) [reply cocos/core/root.ts]: ok

Comment (minggo) [inline cocos/core/root.ts]: Why put window id at first parameter. Does widow id exist on web platform?

Comment (freeboy) [reply cocos/core/root.ts]: I'll change it to the last optional paremeter.

**Comment by freeboy** [comment]
@cocos-robot run test cases

**Review by minggo — APPROVED** [review]

**Comment by freeboy** [comment]
@cocos-robot run test cases

**Review by qiuguohua — APPROVED** [review]

**Review by PPpro — APPROVED** [review]

**Pull request merged**